### PR TITLE
Fixes option pattern matches

### PR DESCRIPTION
**Pull request by nvollmar**
For 1.1.0

### CODE REVIEW

**Comment by mdedetrich** [comment]
@nvollmar Do want to rebase PR to resolve conflicts with main?

**Comment by nvollmar** [comment]
I'll rebase it once we're ready to merge it. So far plan was to do it in 1.1

File: core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala [hunk 1]
```diff
@@ -800,9 +800,8 @@ import akka.stream.scaladsl.Source
           Extractors.optionalTaggedPersistentRepr(eventDeserializer, serialization))
         .mapAsync(1) { t =>
           t.tagged match {
-            case OptionVal.Some(tpr) =>
-              tr.sendMissingTagWrite(tp)(tpr)
             case OptionVal.None => FutureDone // no tags, skip
+            case tpr            => tr.sendMissingTagWrite(tp)(tpr.x)
```
Comment (He-Pin) [inline core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala:804]: Why not just change to :
```scala
case OptionVal.Some(tpr) =>
              tr.sendMissingTagWrite(tp)(tpr.x)
```
?

Comment (nvollmar) [reply core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala]: That would still cause the pattern match warning.

**Review by pjfanning — APPROVED** [review]
lgtm

**Pull request merged**